### PR TITLE
Dropdown — Add option to make the dropdown menu clickable

### LINKED ISSUE #32723: v5: Dropdown - create a class/data-attribute selector to prevent closing dropdown-menu on click
By default `.dropdown-menu` hides when you click inside it. The only exception is when you have a form - it stays open when you focus on input or click buttons. It would be good to have a dedicated class `.dropdown-menu-something` or data attribute `data-something="something"` to prevent `.dropdown-menu` from hiding for content other than forms.

`<ul class="dropdown-menu dropdown-menu-something" aria-labelledby="dropdownMenuButton">`

OR

`<ul class="dropdown-menu" data-something="something" aria-labelledby="dropdownMenuButton">`

It looks like the javascript for preventing such behaviour in forms is very simple so why not add another selector?
Line 634: https://github.com/twbs/bootstrap/blob/main/js/dist/dropdown.js#L634

### PR DESCRIPTION
**Pull request by rohit2sharma95**
Closes #32723

Adding one more option `autoClose` to configure the auto close behavior of the dropdown.

Preview: https://deploy-preview-33389--twbs-bootstrap.netlify.app/docs/5.0/components/dropdowns/#auto-close-behavior
Ref: https://ng-bootstrap.github.io/#/components/dropdown/api#NgbDropdown

### CODE REVIEW

**Comment by mdo** [comment]
I'd love for this and the base PR to be a part of Beta 3 if we can manage it. Offering more control over the dropdown clicks here will be a huge win for usage and general UX.

**Comment by mdo** [comment]
Do we need to do anything for v4? https://github.com/twbs/bootstrap/issues/26189

**Comment by rohit2sharma95** [comment]
It can be backported to V4.

File: js/tests/unit/dropdown.spec.js [hunk 1]
```diff
@@ -1783,4 +1783,51 @@ describe('Dropdown', () => {
 
     triggerDropdown.dispatchEvent(keydown)
   })
+
+  it('should allow `data-bs-toggle="dropdown"` click events to bubble up', () => {
+    fixtureEl.innerHTML = [
+      '<div class="dropdown">',
+      '  <button class="btn dropdown-toggle" data-bs-toggle="dropdown">Dropdown</button>',
```
Comment (GeoSot) [inline js/tests/unit/dropdown.spec.js:1790]: you may can add a `onclick=execsomething()`
```js
execsomething = ( ) => {
   expectations
   done()
}

```

**Comment by mdo** [comment]
What's our next steps here?

**Comment by rohit2sharma95** [comment]
@mdo, Sorry for the delay, I will update it further.

**Comment by XhmikosR** [comment]
@rohit2sharma95 can you take care of this soon? Otherwise let's move it to 5.1.0.

**Review by alpadev — APPROVED** [review]

**Review by GeoSot — APPROVED** [review]

**Comment by rohit2sharma95** [comment]
Should it be backported to V4? @mdo @XhmikosR

**Comment by XhmikosR** [comment]
@rohit2sharma95 I guess we could, if it's not too much trouble, just note that this technically seems like a new feature. So, technically it should land in a 4.7.0 version. Just make sure you add the needed labels and we'll see when time comes to release a new 4.x version. :)

**Pull request merged**